shell/platform/darwin/ios/framework/Source/FlutterDartVMServicePublisher.mm

```diff
@@ -139,32 +141,30 @@ static void DNSSD_API RegistrationCallback(DNSServiceRef sdRef,
 
 @implementation FlutterDartVMServicePublisher {
   flutter::DartServiceIsolate::CallbackHandle _callbackHandle;
-  std::unique_ptr<fml::WeakPtrFactory<FlutterDartVMServicePublisher>> _weakFactory;
 }
 
 - (instancetype)initWithEnableVMServicePublication:(BOOL)enableVMServicePublication {
   self = [super init];
   NSAssert(self, @"Super must not return null on init.");
 
-  _delegate.reset([[DartVMServiceDNSServiceDelegate alloc] init]);
+  _delegate = [[DartVMServiceDNSServiceDelegate alloc] init];
   _enableVMServicePublication = enableVMServicePublication;
-  _weakFactory = std::make_unique<fml::WeakPtrFactory<FlutterDartVMServicePublisher>>(self);
+  __weak __typeof(self) weakSelf = self;
 
   fml::MessageLoop::EnsureInitializedForCurrentThread();
 
   _callbackHandle = flutter::DartServiceIsolate::AddServerStatusCallback(
-      [weak = _weakFactory->GetWeakPtr(),
-       runner = fml::MessageLoop::GetCurrent().GetTaskRunner()](const std::string& uri) {
+      [weakSelf, runner = fml::MessageLoop::GetCurrent().GetTaskRunner()](const std::string& uri) {
         if (!uri.empty()) {
-          runner->PostTask([weak, uri]() {
+          runner->PostTask([weakSelf, uri]() {
             // uri comes in as something like 'http://127.0.0.1:XXXXX/' where XXXXX is the port
             // number.
-            if (weak) {
-              NSURL* url = [[[NSURL alloc]
-                  initWithString:[NSString stringWithUTF8String:uri.c_str()]] autorelease];
-              weak.get().url = url;
-              if (weak.get().enableVMServicePublication) {
-                [[weak.get() delegate] publishServiceProtocolPort:url];
+            if (weakSelf) {
```

Comment:
I'm surprised we aren't getting a warning here; this is violating the repeated-use-of-weak-self that Xcode usually warns on. The strongly preferred pattern is to have the first statement be a `strongSelf = weakSelf`, and then have all the instances be `strongSelf`, so that it's impossible to have `self` become nil part way through the method.